File: modules/native-audio/android/src/main/java/expo/modules/nativeaudio/TimelinePlaybackEvent.kt
package expo.modules.nativeaudio

/**
 * One compiled score tick passed from SongPlaybackCompiler (JS).
 * Timing is derived from [bpm] and [ticksPerBeat] when building offsets.
 */
internal data class TimelinePlaybackEvent(
  val sequence: Long,
  val bpm: Double,
  val accent: Boolean,
  val subdivisionIndex: Int,
  val ticksPerBeat: Int,
  val beatIndexInBar: Int,
  val beatsPerMeasure: Int,
  val barId: String,
  val sectionId: String,
)

File: modules/native-audio/ios/TimelinePlaybackEvent.swift
import Foundation

/// One compiled score tick from SongPlaybackCompiler (JS wire format).
struct TimelinePlaybackEvent {
  let sequence: UInt64
  let bpm: Double
  let accent: Bool
  let subdivisionIndex: Int
  let beatIndexInBar: Int
  let beatsPerMeasure: Int
  let barId: String
  let sectionId: String
}
